### PR TITLE
Feature/layout

### PR DESCRIPTION
**Pull request by sebastianmusial**
DONE:
- layout store
- RWD styles (global spaces)
- approach - utilities styles in scss (only component classes). This approach gives easier styles overwriting (in RWD) and clearer code.
- mobile navigation with a toggle option
- mobile navigation - close navigation action for items without children
- desktop navigation - hide/show option

### CODE REVIEW

File: src/app/layout/store/layout.reducers.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export const initialState: LayoutState = {
+  isSidenav: false,
+};
+
+export const reducer = (state = initialState, action: layoutActions.All): LayoutState => {
+  switch (action.type) {
+    case layoutActions.LAYOUT_SIDENAV_TOGGLE:
+      return {
+        ...state,
+        isSidenav: action.state === undefined
```
Comment (pdomaleczny) [inline src/app/layout/store/layout.reducers.ts]: I would move this condition check before return statement :D

File: src/app/layout/components/sidenav/sidenav.component.scss [hunk 2]
```diff
@@ -11,7 +34,10 @@
 }
 
 .sidenav__logo {
-  max-height: 60px;
-  margin: 10px;
-  padding: 10px;
+  @extend %u-distance--regular;
+  max-height: 30px;
```
Comment (JozefFlakus) [inline src/app/layout/components/sidenav/sidenav.component.scss]: can we use here linear interpolation?

File: src/app/layout/store/layout.reducers.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { LayoutState } from './layout.state';
+import * as layoutActions from './layout.actions';
+
+export const initialState: LayoutState = {
+  isSidenav: false,
```
Comment (JozefFlakus) [inline src/app/layout/store/layout.reducers.ts]: I prefer `isSidenav` -> `sidenavActive`. `isSidenav` can be confusing - `is sidenav` what?

File: src/app/layout/store/layout.state.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface LayoutState {
+  isSidenav: boolean;
```
Comment (JozefFlakus) [inline src/app/layout/store/layout.state.ts]: see: https://github.com/baristaJS/barista-admin/pull/14/files#r158638396

File: src/app/store/index.ts [hunk 5]
```diff
@@ -61,7 +66,7 @@ export const routerStateProvider = {
 //  Modules
 // ----------------------------------------
 export const storeModule = [
-  StoreModule.forRoot(reducerToken, { initialState })
+  StoreModule.forRoot(reducerToken)
```
Comment (JozefFlakus) [inline src/app/store/index.ts]: Why did you remove `{ initialState }`? It is here on purpose - @ngrx needs to have it setup explicit.

File: src/app/layout/store/layout.reducers.ts (hunk 3 above)
Comment (JozefFlakus) [reply src/app/layout/store/layout.reducers.ts]: It is required by `AoT Compiler`

File: src/app/store/index.ts (hunk 5 above)
Comment (JozefFlakus) [reply src/app/store/index.ts]: It is required by `AoT Compiler`. See: https://github.com/ngrx/platform/blob/master/docs/store/api.md#initial-state-and-ahead-of-time-aot-compilation
